### DISCUSSION TITLE
422 Target Validation Failed Error When Creating A Ruleset with targeted branches

### DISCUSSION THREAD

**Discussion by lado936** [community/community#119797] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hello, i am using go sdk to create Repository with Rulesets, while Ruleset creation is working okay, if i set target branches like "main" "test", i get below error
`
DEBUG   events  PUT https://api.github.com/repos/test/test/rulesets/6545754: 422 Validation Failed [{Resource: Field: Code: Message:Invalid target patterns: 'test', 'main'}] 
 `
 after many trials and errors, i tried to set branches not like main, but as "refs/heads/main", this one worked, but issue is that now in repo settings i see targeted branches are set like "refs/heads/refs/heads/main".
 So either API should allow setting branch patterns without refs/heads/ prefix, or it should not be appended after i am setting it on my side
<img width="904" alt="Screenshot 2024-04-16 at 17 14 32" src="https://github.com/community/community/assets/13964808/c6d66016-e23f-4b1d-b5a5-46e46c01ed4b">

**Comment by jimethn** [comment]
Hey @lado936 , I noticed you resolved this, was there something you needed to do to get it working? I'm finding a similar issue trying to create rulesets with the github terraform provider. It allows me to set an include pattern of "master" via the UI, but when I try via the API I get

> 422 Validation Failed [{Resource: Field: Code: Message:Invalid target patterns: 'master'}]

**Reply by pkuczynski** [reply]
I have the same problem. @jimethn @lado936 how did you solved it?

**Reply by jchancellor-ms** [reply]
@pkuczynski - I ran into the same error and got it to work with the following config. 

`conditions {
    ref_name {
      include = ["~DEFAULT_BRANCH", "refs/heads/main"]
      exclude = []
    }
  }`

**Reply by pkuczynski** [reply]
I ended up using `"refs/heads/master"` too... Quite annoying since the UI allow you to type just a branch name...
